### PR TITLE
Global Adapter related Implementation

### LINKED ISSUE #2106: Connecting to Global Adapter and fetch APIs via XDS
### Describe your problem(s)
For the choreo deployment, the global adapter will be the component that provides information on which APIs to pull. For that it is required to implement XDS client within adapter component.

### How will you implement it

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Purpose
This includes the connecting to the global adapter and fetching the APIs from global adapter.

### Issues

Fixes #2106 
Fixes part of https://github.com/wso2-enterprise/choreo/issues/4332

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by VirajSalaka** [comment]
comments mentioned in https://github.com/wso2/product-microgateway/pull/2107 are resolved.

File: adapter/config/types.go [hunk 1]
```diff
@@ -62,11 +62,12 @@ func NewReceiver() chan string {
 // Config represents the adapter configuration.
 // It is created directly from the configuration toml file.
 type Config struct {
-	Adapter      adapter
-	Envoy        envoy `toml:"router"`
-	Enforcer     enforcer
-	ControlPlane controlPlane `toml:"controlPlane"`
-	Analytics    analytics    `toml:"analytics"`
+	Adapter       adapter
+	Envoy         envoy `toml:"router"`
```
Comment (Jayanie) [inline adapter/config/types.go:66]: Formatting issue

File: enforcer/src/main/resources/Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 # -----------------------------------------------------------------------
 
-FROM adoptopenjdk/openjdk11:jre-11.0.11_9-alpine
+FROM adoptopenjdk/openjdk11:jdk-11.0.11_9-alpine
```
Comment (mpmunasinghe) [inline enforcer/src/main/resources/Dockerfile:17]: Why do we switch to jdk here?

**Review by mpmunasinghe — APPROVED** [review]
LGTM

File: enforcer/src/main/resources/Dockerfile (hunk 2 above)
Comment (VirajSalaka) [reply enforcer/src/main/resources/Dockerfile:17]: this is introduced from the main branch. So let's keep it as it is, until the main branch is fixed.

File: adapter/internal/adapter/adapter.go [hunk 3]
```diff
@@ -28,6 +28,7 @@ import (
 	"github.com/wso2/product-microgateway/adapter/internal/auth"
 	enforcerCallbacks "github.com/wso2/product-microgateway/adapter/internal/discovery/xds/enforcercallbacks"
 	routercb "github.com/wso2/product-microgateway/adapter/internal/discovery/xds/routercallbacks"
+	"github.com/wso2/product-microgateway/adapter/internal/ga"
```
Comment (NipunaPrashan) [inline adapter/internal/adapter/adapter.go:31]: Shall we keep the name as **global-adapter**

Comment (VirajSalaka) [reply adapter/internal/adapter/adapter.go:31]: https://golang.org/doc/effective_go#package-names

File: adapter/config/types.go (hunk 1 above)
Comment (VirajSalaka) [reply adapter/config/types.go]: fixed

**Review by Jayanie — APPROVED** [review]

File: resources/conf/config.toml.template [hunk 4]
```diff
@@ -364,6 +364,15 @@ idleTimeoutInSeconds = 3600
...
     eventListeningEndpoints = ["amqp://admin:$env{cp_admin_pwd}@apim:5672?retries='10'&connectdelay='30'"]
 
+# Global Adapter related configurations
+[globalAdapter]
+  # Enable Choreo Connect Global adapter client
+  enabled = false
+  # Global adapter Service URL (<Hostname>:<Port>)
+  serviceUrl = "ga:18002"
+  # Local Adapter Label
+  localLabel = "$env{LocalLabel}"
```
Comment (NipunaPrashan) [inline resources/conf/config.toml.template:374]: Shall we keep env variable names to a constant pattern like we have used in this file.
Either **local_label** or **LOCAL_LABEL**

Comment (VirajSalaka) [reply resources/conf/config.toml.template]: fixed using local_label

**Review by NipunaPrashan — APPROVED** [review]
LGTM

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20210825.10

**Comment by choreo-cicd** [comment]
[succeeded] prod-deployment : 20210831.1

**Pull request merged**